api/api/controllers/search_controller.py

```diff
@@ -496,39 +496,39 @@ def search(
     return results, page_count, result_count, search_context.asdict()
 
 
-def related_media(uuid: str, index: str, filter_dead: bool) -> tuple[list[Hit], int]:
-    """Given a UUID, find related search results."""
+def related_media(uuid: str, index: str, filter_dead: bool) -> list[Hit]:
+    """Given a UUID, find related search results based on title and tags."""
 
-    search_client = Search(index=index)
+    # Search the default index for the item itself as it might be sensitive.
+    item_search = Search(index=index)
 
-    # Convert UUID to sequential ID.
-    item = search_client
-    item = item.query("match", identifier=uuid)
-    _id = item.execute().hits[0].id
+    # Get the item's title and tags.
+    item_hit = item_search.query("match", identifier=uuid).execute().hits[0]
+    title = item_hit.title
+    tags = ",".join([tag.name for tag in item_hit.tags])
 
-    s = search_client
-    s = s.query(
-        MoreLikeThis(
-            fields=["tags.name", "title"],
-            like={"_index": index, "_id": _id},
-            min_term_freq=1,
-            max_query_terms=50,
-        )
-    )
-    # Prevent the items that users set as `mature` from showing up in
-    # recommendations.
-    s = s.exclude("term", mature=True)
+    # Search the filtered index for related items.
+    s = Search(index=f"{index}-filtered")
+
+    # Match the title or tags
+    title_query = SimpleQueryString(query=title, fields=["title"])
+    tags_query = SimpleQueryString(fields=["tags.name"], query=tags)
+    related_query = title_query | tags_query
+
+    # Exclude the current item and mature content.
+    # TODO: remove `__keyword` after
+    #  https://github.com/WordPress/openverse/pull/3143 is merged.
+    s = s.query(related_query & ~Match(identifier__keyword=uuid) & ~Term(mature=True))
```

Comment:
Is the `& ~` in `s.query()` equivalent to the previous `s.exclude()`?

Reply:
I _think_ ~ is converted to `must_not` query. Here's the elasticsearch_dsl code for `invert`ing the query:

https://github.com/elastic/elasticsearch-dsl-py/blob/82418c01b4b0350636877e81e105164b8fcf289c/elasticsearch_dsl/query.py#L178-L196

So, the query is excluding identifier match (so, the item itself) and `mature=True`.
The `and` means that all of the clauses must match.